### PR TITLE
Allow limiting the count of monsters in a randomly generated encounter

### PR DESCRIPTION
**Pull request by stuartlangridge**
Add a 'total monster count', to allow for requesting random encounters with only 1 monster, or only 2 monsters, etc. Useful to find one interesting encounter for a party against one big monster, rather than getting a collection of smaller weaker monsters (even if that might be a more challenging encounter).

### CODE REVIEW

**Comment by TheInsomnolent** [comment]
This is wicked. Love the idea.

This + picking a location and a difficulty really is a perfect random encounter tool

File: app/encounter-builder/current-encounter.html [hunk 1]
```diff
@@ -15,6 +15,8 @@ <h2>
 		</ul>
 	</div>
 </h2>
+	<p>No more than <input class="current-encounter--total-monsters form-control input-sm"
```
Comment (Asmor) [inline app/encounter-builder/current-encounter.html]: Improperly formatted. See p tag immediately below for example.

File: app/encounter-builder/current-encounter.controller.js [hunk 2]
```diff
@@ -15,12 +15,14 @@
         vm.isPool = vm.encounter.type == 'pool';
         vm.newEncounter = newEncounter;
         vm.partyInfo = partyInfo;
+        vm.totalMonsters = 10;
 
         var lastDifficulty = "medium";
         
         function generateRandom(difficulty) {
+            console.log("gen random", vm.total_monsters);
```
Comment (Asmor) [inline app/encounter-builder/current-encounter.controller.js]: remove

File: app/services/randomencounter.service.js [hunk 3]
```diff
@@ -65,8 +66,14 @@
 					[ 2, 2 ],
 					[ 2, 4 ],
 					[ 8 ],
-				],
-				groups = JSON.parse(JSON.stringify(templates[Math.floor(Math.random() * templates.length)])),
+				];
+			if (maxMonsters) {
+				templates = templates.filter(function(t) {
+					var sum = t.reduce(function (a, b) { return a+b; });
```
Comment (Asmor) [inline app/services/randomencounter.service.js]: `var` is inappropriate here, as it's function-scoped. This should be `let`.

File: app/services/randomencounter.service.js [hunk 4]
```diff
@@ -65,8 +66,14 @@
...
+					return sum <= maxMonsters;
+				})
```
Comment (Asmor) [inline app/services/randomencounter.service.js]: missing semi-colon

**Comment by Asmor** [comment]
Some corrections are needed. I've left comments in the diff.

**Comment by stuartlangridge** [comment]
I've pushed minor fixes to each of those comments, I think; let me know if they're not right!

**Comment by Asmor** [comment]
This is now merged and live. Thanks!

**Pull request merged**